src/irmin/type/type_core.ml

```diff
@@ -16,8 +16,8 @@
 
 let check_valid_utf8 str =
   Uutf.String.fold_utf_8
-    (fun _ _ -> function `Malformed _ -> invalid_arg "Malformed UTF-8"
-      | _ -> ())
+    (fun _ _ -> function `Malformed _ -> invalid_arg "Malformed UTF-8" | _ -> ()
+      )
```

Comment:
While potentially quite hard to implement, I preferred the previous case here, and would happily consume an extra line to avoid this hanging paren.

```suggestion
    (fun _ _ -> function
      | `Malformed _ -> invalid_arg "Malformed UTF-8" 
      | _ -> () )
```